rllib/examples/inference_and_serving/policy_inference_after_training_with_dt.py

```diff
@@ -0,0 +1,166 @@
+            model={
+                "max_seq_len": 3,
+            },
+            num_layers=1,
+            num_heads=1,
+            embed_dim=64,
+        )
+        # Need to do evaluation rollouts for stopping condition.
+        .evaluation(
+            evaluation_interval=1,
+            evaluation_num_workers=1,
+            evaluation_duration=10,
+            evaluation_duration_unit="episodes",
+            evaluation_parallel_to_training=False,
+            evaluation_config={"input": "sampler", "explore": False},
+        )
+        .rollouts(
+            num_rollout_workers=0,
+            # This needs to be specified
+            horizon=max_ep_len,
+        )
+        .reporting(
+            min_time_s_per_iteration=10,
+        )
+        .resources(
+            # Use GPUs iff `RLLIB_NUM_GPUS` env var set to > 0.
+            num_gpus=int(os.environ.get("RLLIB_NUM_GPUS", "0")),
+        )
+    )
+    config = config.to_dict()
+
+    # Configure when to stop training
+    # Note that for an offline RL algorithm, we don't do training rollouts,
+    # instead we have to rely on evaluation rollouts.
+    stop = {
+        "evaluation/episode_reward_mean": 200.0,
+        "training_iteration": 100,
+    }
+
+    print("Training policy until desired reward/iterations. ...")
+    tuner = tune.Tuner(
+        "DT",
+        param_space=config,
+        run_config=air.RunConfig(
+            stop=stop,
+            verbose=Verbosity.V3_TRIAL_DETAILS,
+            checkpoint_config=air.CheckpointConfig(
+                checkpoint_frequency=1,
+                checkpoint_at_end=True,
+            ),
+        ),
+    )
+    results = tuner.fit()
+
+    print("Training completed. Restoring new Algorithm for action inference.")
+    # Get the last checkpoint from the above training run.
+    checkpoint = results.get_best_result().checkpoint
+    # Create new Algorithm and restore its state from the last checkpoint.
+    algo = get_algorithm_class("DT")(config=config)
+    algo.restore(checkpoint)
+
+    # Create the env to do inference in.
+    env = gym.make("CartPole-v0")
+
+    obs = env.reset()
+    input_dict = algo.get_initial_input_dict(obs)
+
+    num_episodes = 0
+    total_rewards = 0.0
+
+    while num_episodes < args.num_episodes_during_inference:
+        # Compute an action (`a`).
+        a, _, extra = algo.compute_single_action(input_dict=input_dict)
+        # Send the computed action `a` to the env.
+        obs, reward, done, _ = env.step(a)
+        # Add to total rewards.
+        total_rewards += reward
+        # Is the episode `done`? -> Reset.
+        if done:
+            print(f"Episode {num_episodes+1} - return: {total_rewards}")
+            obs = env.reset()
+            input_dict = algo.get_initial_input_dict(obs)
+            num_episodes += 1
+            total_rewards = 0.0
+        # Episode is still ongoing -> Continue.
+        else:
+            input_dict = algo.get_next_input_dict(
```

Comment:
I love these util functions. I think they should be part of all algos.

Reply:
:)))